devel/nodejs21/Portfile

```diff
@@ -46,14 +43,42 @@ depends_build-append    port:pkgconfig \
 depends_lib-append      path:lib/pkgconfig/icu-uc.pc:icu \
                         port:zlib
 
-# error: 'atomic_load<v8::internal::OwnedVector<const unsigned char> >' is unavailable: introduced in macOS 10.9
-set min_darwin 13
-if { ${os.major} < ${min_darwin} } {
-    known_fail yes
-    pre-fetch {
-        ui_error "${name} requires OSX 10.[expr ${min_darwin} - 4] or newer."
-        return -code error "Unsupported OSX version"
-    }
+
+if { ${os.platform} eq "darwin" && ${os.major} < 11 } {
+    # on macOS nodejs only builds against libc++
+    # this force is OK as node does not link against any other c++ libs
+    # per @kencu, libcxx is only for 10.5 and 10.6 as a prerequisite for an even-newer libcxx (macports-libcxx)
+    depends_lib-append      port:libcxx
+    depends_lib-append    path:lib/libuv.dylib:libuv
+    configure.args-append --shared-libuv
+}
+
+#apparently even macos14 needs the flags that i originally included for older macs.
+#so i am going to make the 'ignore warnings as errors' applicable for all targets
+configure.cppflags-append   -Wno-incompatible-function-pointer-types -Wno-enum-constexpr-conversion
+configure.cflags-append     -Wno-incompatible-function-pointer-types -Wno-enum-constexpr-conversion
+configure.cxxflags-append   -Wno-incompatible-function-pointer-types -Wno-enum-constexpr-conversion
+
+if { ${os.platform} eq "darwin" && ${os.major} < 15 } {
+    # macports-libcxx is required for all platforms below 10.13, i think.
+    # set it to brosemite and lower (14). increase to 18 if it indeed is up to high sierra.
+    depends_lib-append          port:macports-libcxx
```

Comment:
As above, this gonna create a disaster if one builds with `libstdc++`. And while that won’t work due to a wrong hack elsewhere in the port, introducing another bug is not the right way.
`legacysupport` only uses this when the runtime is `libc++`. Even if for some reason this should indeed be done manually (which is probably not the case), it should be conditional on cxx stdlib being `libc++`.